pkg/codegen/python/gen.go

```diff
@@ -1377,6 +1377,10 @@ func (mod *modContext) genResource(res *schema.Resource) (string, error) {
 		fmt.Fprintf(w, "    warnings.warn(\"\"\"%s\"\"\", DeprecationWarning)\n\n", escaped)
 	}
 
+	fmt.Fprintln(w)
+	fmt.Fprintf(w, "    pulumi_type = \"%s\"\n", res.Token)
```

Comment:
Should we emit this for provider resources or only Custom/Component resources?

Reply:
Provider resources still have a type token, and you can get it via the instance based pulumiResourceType methods (because provider resources are still resources) so seems consistent to emit the static type token as well.